docs/src/main/asciidoc/config-your-quarkus-applications-using-a-yaml-file_howto.adoc

```diff
@@ -0,0 +1,407 @@
+
+.Prerequisites
+
+* You have a {project-name} Maven project.
+* You have a PostgreSQL data source.
+* You have the following extensions as dependencies in the `pom.xml` file of your project:
+** `quarkus-rest-client`
+** `quarkus-jdbc-postgresql`
+** `quarkus-config-yaml`
+
+.Procedure
+
+. Open the `src/main/resources/application.yaml` configuration file.
+
+. Add the nested class configuration properties to your `application.yaml` file, as shown in the following example:
++
+.Example `application.yaml` file
+[source,yaml]
+----
+# Properties that configure the JDBC data source driver of your PostgreSQL data source
+quarkus:
+  datasource:
+    url: jdbc:postgresql://localhost:5432/some-database
+    driver: org.postgresql.Driver
+    username: quarkus
+    password: quarkus
+
+# Property that configures the URL of the endpoint to which the rest client sends requests
+org:
+  acme:
+    restclient:
+      CountriesService/mp-rest/url: https://restcountries.eu/rest
+
+# Property that configures the log message level for your application
+quarkus:
+  log:
+    category:
+
+# Do not use spaces in names of configuration properties that you place inside quotation marks
+      "io.quarkus.category":
+        level: INFO
+----
++
+Similarly to using the comments in the `application.properties` file, you can use comments to describe your configuration properties in YAML format.
++
+[NOTE]
+====
+Always use spaces to indent the properties in your YAML configuration file.
+YAML does not support using tabs for indentation.
+====
+
+// CREATE include::modules/conc_configuration-profiles-with-yaml.adoc[leveloffset=+2]
+[id="conc_configuration-profiles-with-yaml"]
+== Configuration profiles with YAML
+
+You can use xref:config-reference.adoc#profiles[profiles] in YAML.
+
+Because YAML does not support names that start with `%`, you must enclose the names of configuration keys between double quotes.
+
+For example, in `application.yaml`, you must use `"%dev"` or `"%test"` as the configuration keys to identify properties that you want to your Quarkus project to use when you build and run it using using the development or test profiles.
+
+.Example `application.yaml` with production and development profiles
+[source, yaml]
+----
+quarkus:
+  http:
+    port: 8081
+
+"%dev":
+    quarkus:
+        http:
+          port: 8082
+----
+
+Quarkus runs the production profile by default. With the preceding example, if you don't specify a profile, the HTTP port number is `8081`. If you enable the development profile, the HTTP port number is `8082`.
+
+The YAML configuration also supports profile-aware files. In this case, properties for a specific profile may reside in
+an `application-{profile}.yaml` named file. The previous example may be expressed as:
+
+.application.yaml
+[source, yaml]
+----
+quarkus:
+  http:
+    port: 8081
+----
+
+.application-development.yaml
+[source, yaml]
+----
+quarkus:
+  http:
+    port: 8082
+----
```

Comment:
You probably mean `application-dev.yaml`.

Reply:
Updated. Thanks.